Run work posted to test synchronization context for up to ten seconds after the test method completes

jnm2 opened the pull request:
Fixes #3209.

rprouse commented on src/NUnitFramework/tests/TestUtils.cs:
```diff
@@ -0,0 +1,22 @@
+using System.Threading;
+using NUnit.Framework.Internal;
+
+namespace NUnit.Framework
```
Should this class be in the folder/namespace `TestUtilities` like the previous class?

jnm2 replied:
Sure, was 50/50 on this.

jnm2 replied:
Ugh, also forgot the license boilerplate.

jnm2 commented on src/NUnitFramework/framework/Internal/SingleThreadedSynchronizationContext.cs:
```diff
@@ -87,35 +109,37 @@ public void ShutDown()
+                    case Status.ShuttingDown:
+                    case Status.ShutDown:
+                        break;
+                }
```
Good question, haha! It's supposed to be `return;`. I'll write a test enforcing this.

jnm2 commented:
@rprouse I added a new commit in which I changed the no-op switch to `return;` and added a new test demonstrating the need. I rebased the other change in. Ready for review.
